src/background/api/index.js

```diff
@@ -1,19 +1,99 @@
-import * as oasis from '@oasisprotocol/client';
+import * as oasis from "@oasisprotocol/client";
 import { cointypes, TX_LIST_LENGTH } from "../../../config";
-import { amountDecimals, isNumber } from '../../utils/utils';
+import { amountDecimals, isNumber } from "../../utils/utils";
 import { commonFetch, getOasisClient } from "./request";
-import * as oasisRT from '@oasisprotocol/client-rt';
-const MAX_LENGTH = 500
+import { getRpcAccount } from "./rpc";
+import * as oasisRT from "@oasisprotocol/client-rt";
+const MAX_LENGTH = 500;
+
+// h/t: https://github.com/oasisprotocol/oasis-wallet-web/blob/a554cf974afc69ea27035d809b3649cde2d350b0/src/vendors/monitor.ts#L157
+function getSharePrice(pool) {
+  const balance = Number(oasis.quantity.toBigInt(pool.balance)) / 10 ** 9;
+  const share = Number(oasis.quantity.toBigInt(pool.total_shares)) / 10 ** 9;
+  return balance / share;
+}
+
 /**
  * get account balance
  */
 export async function getBalance(address) {
-  let url = "/chain/account/info/" + address
-  let account = await commonFetch(url).catch(() => { })
+  let url = "/chain/account/info/" + address;
+  let account = await commonFetch(url).catch(() => {});
   if (account && account.code === 0) {
-    return { ...account.data, address }
+    return { ...account.data, address };
   } else {
-    return {}
+    let account = await getRpcAccount(address);
+    const balanceAmount = account?.general?.balance || 0;
+    let balance = balanceAmount;
+    if (balance) {
+      balance = oasis.quantity.toBigInt(balance).toString();
+    }
+
+    const oasisClient = getOasisClient();
+    let shortKey = await oasis.staking.addressFromBech32(address);
+    let height = oasis.consensus.HEIGHT_LATEST;
+
+    let debonding = await oasisClient
+      .stakingDebondingDelegationInfosFor({ height: height, owner: shortKey })
+      .catch((err) => err);
+
+    let delegation = await oasisClient
+      .stakingDelegationInfosFor({ height: height, owner: shortKey })
+      .catch((err) => err);
+
+    const delegationArr = Array.from(delegation, ([addr, delegationInfo]) => ({
+      addr,
+      delegationInfo,
+    }));
+    let escrowTotal = 0;
+    for (const { /*addr: bytesAddress,*/ delegationInfo: d } of delegationArr) {
+      /*const strAddr = oasis.address.toBech32("oasis", bytesAddress);*/
```

Comment:
Not sure if this was an intentional change

escrowTotal uses `for of`, debondingTotal uses `Array.reduce`

Reply:
It was intentional. I had to first unpack `delegation` like this
```
  const delegationArr = Array.from(delegation, ([addr, delegationInfo]) => ({
    addr,
    delegationInfo,
  }));
  ```

Reply:
I think the issue here is consistency.

Both, escrowTotal and debondingTotal should use the same variant. Also, I don't think unpacking is necessary.